### DISCUSSION TITLE
Content modification during io::copy

### DISCUSSION THREAD

**Discussion by alishir** [tokio-rs/tokio#3725] [Q&A] [answered]
Hi all,
Sorry for the beginner's question, I want to modify the content of a byte stream during `io::copy`, for example replacing all `A` characters with `B`. Should I implement custom Future such as Copy?

**Comment by Darksonn** [comment, marked as answer]
You do not necessarily need to implement it with a custom `Future` like `io::copy` does, but it is certainly an option. Some other options are:

 1. An `io::copy` like utility could also be written with async/await syntax using a loop or `tokio::select!`.
 2. You can implement an `AsyncRead` wrapper that performs the conversion and use it with the ordinary `io::copy`.
